src/components/application_manager/test/rpc_service_impl_test.cc

```diff
@@ -301,12 +331,362 @@ TEST_F(RPCServiceImplTest,
-  rpc_service_->SendMessageToMobile(message);
+TEST_F(RPCServiceImplTest,
+       IsAppServiceRPC_SourceMobileWithRightFuncId_ReturnTrue) {
+  auto source_mobile = am::commands::Command::CommandSource::SOURCE_MOBILE;
+  auto source_sdl = am::commands::Command::CommandSource::SOURCE_SDL;
+
+  EXPECT_TRUE(rpc_service_->IsAppServiceRPC(
+      mobile_apis::FunctionID::GetSystemCapabilityID, source_mobile));
+  EXPECT_TRUE(rpc_service_->IsAppServiceRPC(
+      mobile_apis::FunctionID::OnSystemCapabilityUpdatedID, source_sdl));
+  EXPECT_TRUE(rpc_service_->IsAppServiceRPC(
+      mobile_apis::FunctionID::GetSystemCapabilityID, source_sdl));
+  EXPECT_TRUE(rpc_service_->IsAppServiceRPC(
+      mobile_apis::FunctionID::OnSystemCapabilityUpdatedID, source_sdl));
+}
+
+TEST_F(RPCServiceImplTest,
+       IsAppServiceRPC_SourceHMIOrSDLToHMIWithRightFuncId_ReturnTrue) {
+  auto source_hmi = am::commands::Command::CommandSource::SOURCE_HMI;
+  auto source_sdl_to_hmi =
+      am::commands::Command::CommandSource::SOURCE_SDL_TO_HMI;
+  int32_t function_id =
+      hmi_apis::FunctionID::BasicCommunication_OnSystemCapabilityUpdated;
+
+  EXPECT_TRUE(rpc_service_->IsAppServiceRPC(function_id, source_hmi));
+  EXPECT_TRUE(rpc_service_->IsAppServiceRPC(function_id, source_sdl_to_hmi));
+}
+
+TEST_F(RPCServiceImplTest, IsAppServiceRPC_PluginIsEmpty_ReturnFalse) {
+  int32_t function_id = kFunctionId;
+  auto source = am::commands::Command::CommandSource::SOURCE_HMI;
+  ON_CALL(mock_app_mngr_, GetPluginManager())
+      .WillByDefault(ReturnRef(mock_rpc_plugin_manager_));
+  typedef am::plugin_manager::RPCPlugin RPCPlugin;
+  utils::Optional<RPCPlugin> mock_rpc_plugin_opt(
+      utils::Optional<RPCPlugin>::OptionalEmpty::EMPTY);
+  ON_CALL(mock_rpc_plugin_manager_, FindPluginToProcess(kFunctionId, source))
+      .WillByDefault(Return(mock_rpc_plugin_opt));
+
+  EXPECT_FALSE(rpc_service_->IsAppServiceRPC(function_id, source));
+}
+
+TEST_F(RPCServiceImplTest, IsAppServiceRPC_PluginWithWrongName_ReturnFalse) {
+  int32_t function_id = kFunctionId;
+  auto source = am::commands::Command::CommandSource::SOURCE_HMI;
+  ON_CALL(mock_app_mngr_, GetPluginManager())
+      .WillByDefault(ReturnRef(mock_rpc_plugin_manager_));
+  utils::Optional<am::plugin_manager::RPCPlugin> mock_rpc_plugin_opt(
+      mock_rpc_plugin_);
+  ON_CALL(mock_rpc_plugin_manager_, FindPluginToProcess(_, _))
+      .WillByDefault(Return(mock_rpc_plugin_opt));
+  ON_CALL(mock_rpc_plugin_, PluginName()).WillByDefault(Return(""));
+  EXPECT_FALSE(rpc_service_->IsAppServiceRPC(function_id, source));
 }
 
+TEST_F(RPCServiceImplTest,
+       IsAppServiceRPC_CommandFactoryIsAbleToProcess_ReturnTrue) {
+  int32_t function_id = kFunctionId;
+  auto source = am::commands::Command::CommandSource::SOURCE_HMI;
+  ON_CALL(mock_app_mngr_, GetPluginManager())
+      .WillByDefault(ReturnRef(mock_rpc_plugin_manager_));
+  utils::Optional<am::plugin_manager::RPCPlugin> mock_rpc_plugin_opt(
```

Comment:
You use `utils::Optional<am::plugin_manager::RPCPlugin>` many tymes, I would typedef it

Reply:
Fixed in commit 9cde89695d5fd433dd02f3d3af926c3bf2f5575b